docs/sphinx/source/contributing/style_guide.rst

```diff
@@ -107,6 +107,107 @@ Note that Windows users need not have the ``make`` utility installed as pvlib
 includes a ``make.bat`` batch file that emulates its interface.
 
 
+.. _example-docstring:
+
+Example Docstring
+-----------------
+
+Here is a template for a function docstring that tries to encapsulate all
+the features that may be used in any docstring. This is a template, so
+not all sections are required for every function, but is a good starting point.
+
+.. code-block:: python
+
+    def example_function(poa_global, exponents, degree_symbol,
+                        default_arg='someval', optional_arg=None):
+        r"""
+        A brief summary of the function.
+
+        A longer description of the function. This can include references
+        to literature [1]_, websites, and other code elements such as
+        functions (:py:func:`pvlib.location.lookup_altitude`) and
+        classes (:py:class:`pvlib.location.Location`).
+
+        .. versionadded:: 0.0.1
+
+        .. note:: This docstring is a template.
+
+        .. warning:: Code body is not included in this example!
+
+        Parameters
+        ----------
+        poa_global : numeric
+            Plane-of-array global irradiance, see :term:`poa_global`. [Wm⁻²].
+        exponents : array-like
+            A list of exponents. [x⁰¹²³⁴⁵⁶⁷⁸⁹⁺⁻].
+        degree_symbol : numeric
+            It's different from superscript zero. [°].
+        default_arg : string, default ``'someval'``.
+            Text enclosed in double backticks mean code formatting.
```

Comment:
```suggestion
        default_arg_1 : string, default ``'someval'``.
            Text enclosed in double backticks mean code formatting.
        time_ref : string, {'UT', 'TST'}, default: 'UT'
            'UT' (universal time) or 'TST' (True Solar Time)
```
How about adding one where there is a set of values you need to choose from?

Reply:
That has never been a standard in pvlib, right? I've usually found the possible values listed in the param description body. Don't get me wrong, I'd love pvlib to have that + type annotations with the possible values. My only remarks are the length when there is a longer list of items and the colon, I find `:` is visual noise.

```suggestion
        default_arg_1 : string, default ``'someval'``.
            Text enclosed in double backticks mean code formatting.
        time_ref : ``'UT'`` or ``'TST'``, default ``'UT'``
            ``'UT'`` (universal time) or ``'TST'`` (True Solar Time)
```

Reply:
Maybe not a standard, but it is used:

```python
pvlib-python/pvlib$ git grep ": {.*}, default"
iotools/pvgis.py:    pvtechchoice: {'crystSi', 'CIS', 'CdTe', 'Unknown'}, default: 'crystSi'
iotools/pvgis.py:    mountingplace: {'free', 'building'}, default: free
iotools/pvgis.py:    trackingtype: {0, 1, 2, 3, 4, 5}, default: 0
iotools/solaranywhere.py:    time_resolution: {60, 30, 15, 5}, default: 60
iotools/solaranywhere.py:    spatial_resolution: {0.1, 0.01, 0.005}, default: 0.01
iotools/solaranywhere.py:    missing_data: {'Omit', 'FillAverage'}, default: 'FillAverage'
iotools/solargis.py:    timestamp_type : {'start', 'center', 'end'}, default: 'center'
pvsystem.py:    fill_method : {'ffill', 'bfill', 'nearest'}, default 'ffill'
```